### PR TITLE
test(sidechain): adds test coverage to sidechain smart contracts

### PR DESCRIPTION
**Pull request by 0xahtle7**

### CODE REVIEW

File: test/convex-platform/BoosterLite.spec.ts [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+const L1_CHAIN_ID = 111;
+const L2_CHAIN_ID = 222;
+
+// ADD TESTS TO VERIFY WRONG STORAGE DUE TO BAD srcs
+//  @method Booster.earmarkRewards: Error: VM Exception while processing transaction: reverted with reason string 'LayerZeroMock: not enough native for fees'
+// @method BaseRewardPool.processIdleRewards():
+// Error: VM Exception while processing transaction: reverted with reason string 'LayerZeroMock: not enough native for fees'
+// @method BaseRewardPool.stake:
+// Error: VM Exception while processing transaction: reverted with reason string 'RewardPool : Cannot stake 0'
+describe("BoosterLite", () => {
```
Comment (phijfry) [inline test/convex-platform/BoosterLite.spec.ts:32]: Should we add some tests for initialise, "Initialise cannot be called twice"

File: test/sidechain/L1Coordinator.spec.ts [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+            const l1CoordinatorAfter = await crv.balanceOf(l1Coordinator.address);
+
+            expect(feeDebtAfter, "settleFeeDebt").to.be.eq(feeDebtBefore.sub(feeDebt));
+            expect(bridgeDelegateAfter, "bridgeDelegate balance").to.be.eq(bridgeDelegateBalance.sub(feeDebt));
+            expect(l1CoordinatorAfter, "l1Coordinator balance").to.be.eq(l1CoordinatorBalance.add(feeDebt));
+        });
+    });
+    describe("Edge cases", () => {
+        describe("distributeAura", async () => {
+            xit("fails fee is distributed but not settle yet", async () => {
```
Comment (phijfry) [inline test/sidechain/L1Coordinator.spec.ts:294]: Fees don't need to be settled before AURA is distributed.

File: test/sidechain/L2Coordinator.spec.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (phijfry) [inline test/sidechain/L2Coordinator.spec.ts:1]: We can probably add a few more tests to this file, call queueNewRewards, mint etc.

File: test/sidechain/AuraBALOFT.spec.ts [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+    const to = sender.address;
+    await token.connect(sender.signer).approve(proxyOFT.address, amount);
+    const tx = await proxyOFT
+        .connect(sender.signer)
+        .sendFrom(from, dstChainId, to, amount, ZERO_ADDRESS, ZERO_ADDRESS, [], {
+            value: NATIVE_FEE,
+        });
+    return tx;
+}
+describe("AuraBalOFT", () => {
```
Comment (phijfry) [inline test/sidechain/AuraBALOFT.spec.ts:42]: Can probably add some tests for `executeVault` here too.

File: test/convex-platform/BoosterLite.spec.ts (hunk 1 above)
Comment (0xahtle7) [reply test/convex-platform/BoosterLite.spec.ts:22]: Done

File: test/sidechain/L2Coordinator.spec.ts (hunk 3 above)
Comment (0xahtle7) [reply test/sidechain/L2Coordinator.spec.ts]: Done

File: test/sidechain/AuraBALOFT.spec.ts (hunk 4 above)
Comment (0xahtle7) [reply test/sidechain/AuraBALOFT.spec.ts:60]: Done

File: test/convex-platform/BaseRewardPool4626.spec.ts [hunk 5]
```diff
@@ -268,6 +286,88 @@ describe("BaseRewardPool4626", () => {
             expect(await crvRewards.maxWithdraw(aliceAddress)).eq(rwdBalanceAfter);
             expect(await crvRewards.maxRedeem(aliceAddress)).eq(rwdBalanceAfter);
         });
+        it("allows transferFrom between accounts", async () => {
```
[the remark this answers is not in the document]
Comment (0xahtle7) [reply test/convex-platform/BaseRewardPool4626.spec.ts]: Partially Done , has a dependency with the other repo , by now the tests are disabled

**Review by phijfry — APPROVED** [review]
LGTM
